src/fsio.c

```diff
@@ -5130,8 +5126,6 @@ int pr_fsio_close(pr_fh_t *fh) {
 
   if (fh->fh_pool != NULL) {
     destroy_pool(fh->fh_pool);
-    fh->fh_pool = NULL;
-    fh->fh_path = NULL;
```

Comment:
The `fh` object here is itself allocated out of the `fh_pool`, thus attempting to dereference the `fh` pointer here, after the call to `destroy_pool()`, is itself a use-after-free case.